### ISSUE TITLE
OSC needs close-box or something

### ISSUE THREAD

**Issue by archenemies** [mpv-player/mpv#3912]
Hi, I just wanted to comment on some new defaults. I just upgraded and noticed that the OSC - either new, or newly on by default - appearing at the bottom of the screen. It was a bit intrusive visually, especially since when I watch movies I tend to leave my mouse dangling off somewhere, and the OSC kept appearing and disappearing. Figuring out how to turn it off was non-trivial, since first of all I didn't know what it's called. I like Mpv because it's simple and usually stays out of my hair. I guess someone must have taken some time to implement the OSC and so you want to have it on by default so that people know about it, but I think we need to do some brainstorming about how to make it easy for people to turn it off, including how to let people find out what it's called without re-reading every line of the documentation. Also, it seems to display whenever my mouse is anywhere in the bottom half of the screen, which seems like an excessively large region - I would have expected it to only appear when my mouse is over the bottom 50 pixels or something. Thanks!

### mpv version and platform

mpv 0.22.0 (C) 2000-2016 mpv/MPlayer/mplayer2 projects
 built on Sat Nov 26 20:40:21 CET 2016
ffmpeg library versions:
   libavutil       55.34.100
   libavcodec      57.64.101
   libavformat     57.56.100
   libswscale      4.2.100
   libavfilter     6.65.100
   libswresample   2.3.100
ffmpeg version: 3.2.2

**Comment by wiiaboo** [comment]
There's [tons of options](https://mpv.io/manual/stable/#configuration) available to customize the OSC to your needs. Or just disable it with `--no-osc`. Go wild.

**Comment by archenemies** [comment]
> I think we need to do some brainstorming about how to make it easy for people to turn it off, including how to let people find out what it's called without re-reading every line of the documentation. Also, it seems to display whenever my mouse is anywhere in the bottom half of the screen, which seems like an excessively large region - I would have expected it to only appear when my mouse is over the bottom 50 pixels or something. Thanks!

**Comment by CounterPillow** [comment]
>Also, it seems to display whenever my mouse is anywhere in the bottom half of the screen, which seems like an excessively large region - I would have expected it to only appear when my mouse is over the bottom 50 pixels or something. Thanks!

It only appears if you move in that region. The default used to be that it only appears if you move it at the very bottom, but many users didn't like this, since it meant shaking the mouse would no longer show you where you were at in the video file, and you needed to move the mouse all the way to the bottom.

It's a matter of taste, and the whole point of OSC configuration is to account for subjective tastes.

**Comment by archenemies** [comment]
Thanks @CounterPillow for the more complete reply.

My tastes are currently accounted for since, as I said, I found out how to turn OSC off.

I submitted this issue because I was thinking about new users - who probably won't even realize that the feature is configurable.

I think that a better interface would be one that eventually hides the OSC even when the mouse is at the very bottom of the screen. Currently it stays open and I think that was the biggest problem for me. I imagine a lot of people have a wireless mouse that gets kicked about during movie watching; my situation is not quite like that, but similar. Obviously, when the mouse gets kicked it's going to tend to end up at one side or the other, and if this results in having to get up out of your recliner to fix it, that'll be super annoying.

I would recommend either putting a 10-pixel space between the OSC and the bottom; or eventually hiding the OSC even when the mouse is over it; or both.

Thanks to you both for your efforts.

**Comment by wiiaboo** [comment]
> I would recommend either putting a 10-pixel space between the OSC and the bottom; or eventually hiding the OSC even when the mouse is over it; or both.

[barmargin=10](https://mpv.io/manual/stable/#on-screen-controller-barmargin) if you're using layout=bottombar or topbar.

**Comment by archenemies** [comment]
@wiiaboo Thanks - it's great that that is configurable; just to be clear, my suggestion was to have that as a default. You can disagree with that, but I think the point needs to be made that not many Mpv users are going to want to spend time tweaking the appearance or behavior of the OSC - most will not even realize it's configurable. So I think it's reasonable to talk about defaults that work well for average users. I can see from your contributions that you've done a lot of work on this, so I can understand you wanting to just hand it off to the users at this stage. My suggestions are not meant as a rejection of your efforts on this project.

**Comment by wiiaboo** [comment]
What if I argue that I like the bottombar without a margin so I can just move the mouse cursor down at will to show the OSC and use the seekbar? Why should my default make more sense than yours or vice-versa?

This is not a committee-decided project. The default now uses the same logic as every other player that has an non-floating OSC (QuickTime/`layout=box`/`layout=slimbox`). We don't have telemetry so we have no way of knowing how many users even use the OSC, or users that have a sofa with a certain feature that makes wireless mice drift a certain way.

**Comment by archenemies** [comment]
@wiiaboo: Your second paragraph makes it sound like you don't value my input and don't want to continue the discussion, but your first paragraph makes it sound like you do. Which is it?

To answer the question in your first paragraph, I was under the impression that, and I think it was clear from the earlier discussion that, moving the mouse anywhere in the lower half of the screen causes the OSC to display in the default configuration. Why it should immediately hide itself when "barmargin" is set and I move it to the very bottom, is not obvious to me and was not the behavior I was advocating - although I now see that this is the current behavior.

I was only making some suggestions to address a simple problem I observed: that leaving the mouse pointer in certain positions causes the OSC to stay visible indefinitely (I said "Currently it stays open and I think that was the biggest problem for me"). And moreover, that the bottom of the screen is such a position, and that the mouse frequently ends up there when I'm not using it. I don't see how this conflicts with your use-case "so I can just move the mouse cursor down at will to show the OSC and use the seekbar". Maybe it's time to re-read what I wrote and think about it for a minute or two.

I didn't anticipate spending so much time on this bug report, but I'm grateful for the software that you are all providing. So if you think this is helping you then I'm glad to keep answering questions. Otherwise let's both get back to what we were doing before.
